### DISCUSSION TITLE
implement performance budgets for query results size

### DISCUSSION THREAD

**Discussion by pieh** [gatsbyjs/gatsby#29869] [Ideas / Feature Requests]
Warn whenever query result is large enough to cause frontend runtime perf implications. "large enough" is yet to be determined - any ideas?

Notes and questions:
 - ~We only run queries that we need to, so getting size of queries that don't need to re-run will eighter need extra I/O or we will need to cache size of queries~ We started to keep track of page-data size in https://github.com/gatsbyjs/gatsby/pull/20146
 - What should be default budget for query result size?
 - How dev can customize default budget? (maybe we don't want that?)

Related Issues:
 - ~[Add to telemetry how big page json & page bundle sizes](https://github.com/gatsbyjs/gatsby/issues/18819)~ Implemented in https://github.com/gatsbyjs/gatsby/pull/20146
 - [`gatsby/cache-dir/loader.js` eval's JSON on main thread, doesn't use `fetch()`](https://github.com/gatsbyjs/gatsby/issues/18787)

**Comment by teclone** [comment]
is it not possible to run this process in a worker process when available.?
